### PR TITLE
Enable EFA at Runtime

### PR DESCRIPTION
**Pull request by sean-smith**
This builds the rpms into the ami, then set the limits at runtime.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: recipes/_efa_enable.rb [hunk 1]
```diff
@@ -13,13 +13,10 @@
...
-remote_file efa_tarball do
-  source node['cfncluster']['efa']['installer_url']
-  mode '0644'
-  retries 3
-  retry_delay 5
-  not_if { ::File.exist?(efa_tarball) }
+bash "enable efa limits" do
+  cwd Chef::Config[:file_cache_path]
+  code <<-EFAENABLE
+    cd aws-efa-installer/install
```
Comment (lukeseawalker) [inline recipes/_efa_enable.rb]: file_cache_path is a temp dir, so aws-efa-installer could be not present anymore, between the base_install and the base_config recipes

Comment (sean-smith) [reply recipes/_efa_enable.rb]: changed to sources dir: `/opt/parallelcluster/sources`

File: recipes/base_install.rb [hunk 2]
```diff
@@ -236,7 +236,14 @@
...
 
-# Fetch EFA Driver
-unless node['cfncluster']['cfn_region'].start_with?("cn-")
-  include_recipe "aws-parallelcluster::_efa_fetch"
+# Install EFA
+if (node['platform'] == 'centos' && node['platform_version'].to_i >= 7) || node['platform'] == 'amazon' || (node['platform'] == 'ubuntu' && node['platform_version'] == "16.04")
+  unless node['cfncluster']['cfn_region'].start_with?("cn-")
+    include_recipe "aws-parallelcluster::_efa_install"
+  else
+    package 'openmpi-devel' do
```
Comment (lukeseawalker) [inline recipes/base_install.rb]: In Ubuntu there is no openmpi-devel, package name is different

Comment (sean-smith) [reply recipes/base_install.rb]: ubuntu will be a separate patch, I've removed it for now

**Pull request merged**